aten/src/ATen/native/mps/operations/AdaptivePooling.mm

```diff
@@ -71,13 +85,37 @@
                     strideH, strideW,
                     kernel_sizeH, kernel_sizeW);
 
-  output =  at::avg_pool2d(input,
-                           IntArrayRef({kernel_sizeH, kernel_sizeW}),
-                           IntArrayRef({strideH, strideW}),
-                           IntArrayRef({0, 0}),
-                           false,
-                           true,
-                           c10::nullopt);
+  if(isizeH >= osizeH) {
+    output =  at::avg_pool2d(input,
+                             IntArrayRef({kernel_sizeH, kernel_sizeW}),
+                             IntArrayRef({strideH, strideW}),
+                             IntArrayRef({0, 0}),
+                             false,
+                             true,
+                             c10::nullopt);
+  }
+
+  else {
+    Tensor phony_grad = at::ones_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+    auto num_input_dims = input.sizes().size();
+    int64_t phony_shape[num_input_dims];
+    for(int i = 0; i < num_input_dims - 2; i++)
+      phony_shape[i] = input.size(i);
+    phony_shape[num_input_dims-2] = output_size[0];
+    phony_shape[num_input_dims-1] = output_size[1];
+    phony_grad.resize_(IntArrayRef(phony_shape, num_input_dims));
+    output =  at::avg_pool2d_backward(input,
```

Comment:
Lets create one graph to do this. If you use aten:: operations it will be different graph. Can you also explain how using the backward pass work for OutputH is greater than inputH?